### PR TITLE
Guide to migrate Zone CP from On-Prem Global CP to Konnect

### PR DESCRIPTION
**Pull request by lobkovilya**
### Description

Added a new guide on how to migrate Zone from on-prem Global to Konnect
Fix https://github.com/Kong/kong-mesh/issues/3503
Preview https://deploy-preview-6320--kongdocs.netlify.app/konnect/mesh-manager/migrate-zone/

### Testing instructions

Preview link: 

### Checklist 

- [X] Review label added 
- [X] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/konnect/mesh-manager/migrate-zone.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+* A [{{site.mesh_product_name}} global control plane in {{site.konnect_short_name}}](/konnect/mesh-manager/service-mesh/#create-a-zone-in-the-global-control-plane)
+* [yq installed](https://github.com/mikefarah/yq)
+
+## Transfer resources from the self-managed global control plane to {{site.konnect_short_name}}
+
+1. Make sure the self-managed global control plane is active:
+  ```bash
+  kumactl config control-planes list
+  ```
+  This should return your self-managed global control plane. You can use port-forward of the On-Prem Global CP API port 5681 to localhost to get quick access with `kumactl`.
```
Comment (cloudjumpercat) [inline app/konnect/mesh-manager/migrate-zone.md:21]: I'm not sure what the second sentence means. Is there a command you can use to make sure the on prem global cp is active on port 5681? Or does this do something else?

**Review by cloudjumpercat — COMMENTED** [review]
@lobkovilya This is a great doc write up! Thanks for adding this. I made some copyedits and left a comment with some questions I had about one sentence, if you wouldn't mind taking a look at that.

File: app/konnect/mesh-manager/migrate-zone.md (hunk 1 above)
Comment (lobkovilya) [reply app/konnect/mesh-manager/migrate-zone.md:21]: If a user runs self-managed global cp on Kubernetes then quite likely they might be not familiar how to operate it with `kumactl` (because they're using `kubectl`). Normally user has to authenticate `kumactl` with API by following the [instruction](https://docs.konghq.com/mesh/latest/production/secure-deployment/api-server-auth). But there is a simpler way and I wanted to add this like a tip – port-forward Global CP API port to localhost.

But I just noticed that the instruction I mentioned above already contains information about the "port-forward". So I think it's safe to remove the second sentence altogether.

File: app/konnect/mesh-manager/migrate-zone.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+title: Migrate a self-managed global zone control plane to {{site.konnect_saas}}
```
Comment (lahabana) [inline app/konnect/mesh-manager/migrate-zone.md:2]: global zone control plane? Seems odd right?

File: app/_data/docs_nav_konnect.yml [hunk 3]
```diff
@@ -118,7 +118,9 @@
       url: /mesh-manager/
     - text: Create a mesh with the Kubernetes demo app
       url: /mesh-manager/service-mesh/
-      
+    - text: Migrate a Self-managed Zone Control Plane to Konnect
```
Comment (lahabana) [inline app/_data/docs_nav_konnect.yml:121]: Title seems wrong

File: app/konnect/mesh-manager/migrate-zone.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+content_type: tutorial
+---
+
+This guide explains how to migrate a self-managed global control plane to {{site.konnect_saas}}. This can be useful if you manage other services in {{site.konnect_saas}} that way you can view all your entities in one place.
```
Comment (lahabana) [inline app/konnect/mesh-manager/migrate-zone.md:6]: Can we get something a bit more selling? It should probably link back to the overview to explain what mesh manager is

File: app/konnect/mesh-manager/migrate-zone.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+title: Migrate a self-managed global zone control plane to {{site.konnect_saas}}
+content_type: tutorial
+---
+
+This guide explains how to migrate a self-managed global control plane to {{site.konnect_saas}}. This can be useful if you manage other services in {{site.konnect_saas}} that way you can view all your entities in one place. 
+
+## Prerequisites
+
+* A universal or Kubernetes cluster with a running zone control plane connected to the self-managed global control plane
+* [The latest version of kumactl](/mesh/latest/production/install-kumactl/) installed and configured to communicate with the zone control plane
```
Comment (lahabana) [inline app/konnect/mesh-manager/migrate-zone.md:11]: With the zone?

File: app/konnect/mesh-manager/migrate-zone.md (hunk 2 above)
Comment (lobkovilya) [reply app/konnect/mesh-manager/migrate-zone.md:2]: Right, should be
```suggestion
title: Migrate a self-managed zone control plane to {{site.konnect_saas}}
```

**Review by lahabana — COMMENTED** [review]
There's a constant confusion between zones and global which makes this guide hard to follow.
Are we trying to switch zones from an existing on-prem global control-plane to a mink one?
There's nowhere atm where we say update the config of the CP and restart the zone CP for example.

Also it would be hard to add limitations. For example we already discussed that this process would cause cross-zone traffic to break for some time.

File: app/konnect/mesh-manager/migrate-zone.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  ```
+
+1. Apply resources fetched from the self-managed global control plane in steps 2 and 3: 
+  ```bash
+  kumactl apply -f {file_name}
+  ```
+
+## Connect the zone control plane to {{site.konnect_short_name}}
+
+1. [Create a zone in the global control plane](/konnect/mesh-manager/service-mesh/#create-a-zone-in-the-global-control-plane).
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply app/konnect/mesh-manager/migrate-zone.md:68]: What do you mean? Why should it exist if Zone CP is not connected yet?

**Comment by lobkovilya** [comment]
> There's a constant confusion between zones and global which makes this guide hard to follow.

I fixed a couple of places you pointed out.

> Are we trying to switch zones from an existing on-prem global control-plane to a mink one?

Yes

> There's nowhere atm where we say update the config of the CP and restart the zone CP for example.

I was thinking `Follow the instructions in the {{site.konnect_short_name}} UI to set up Helm and a secret token.` should cover that. But you're right, it's better to mention config update and CP restart explicitly. 

> Also it would be hard to add limitations. For example we already discussed that this process would cause cross-zone traffic to break for some time.

Why it would be hard? I can probably add the section at the bottom of the page no?

File: app/konnect/mesh-manager/migrate-zone.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+---
+title: Migrate a self-managed zone control plane to {{site.konnect_saas}}
+content_type: tutorial
+---
+
+This guide explains how to migrate a self-managed global control plane to {{site.konnect_saas}}. This can be useful if you manage other services in {{site.konnect_saas}} that way you can view all your entities in one place.
```
Comment (cloudjumpercat) [inline app/konnect/mesh-manager/migrate-zone.md:6]:
```suggestion
If you already have zone control planes in {{site.mesh_product_name}}, you can migrate them to {{site.konnect_short_name}} in Mesh Manager. [Mesh Manager](/konnect/mesh-manager/) allows you to create, manage, and view your {{site.mesh_product_name}} service meshes using the {{site.konnect_short_name}} platform.

Here are a few benefits of creating a mesh deployment in {{site.konnect_short_name}} instead of using a self-managed setup:

* **Kong-managed global control plane:** By creating your mesh in {{site.konnect_short_name}}, your global control plane is managed by Kong. 
* **All entities in one place:** You can view all your information, such as entities from Kong Ingress Controller (KIC) for Kubernetes, {{site.konnect_short_name}}-managed entities, and now service mesh data, all from one central platform. 
* **Managed UI wizard setup:** {{site.konnect_short_name}} simplifies the process of creating a mesh by providing a setup wizard in the UI that guides you through the configuration steps.

This guide explains how to migrate a self-managed zone control plane to {{site.konnect_saas}}. 
```

@lahabana Does this sound better?

File: app/konnect/mesh-manager/migrate-zone.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+* A [{{site.mesh_product_name}} global control plane in {{site.konnect_short_name}}](/konnect/mesh-manager/service-mesh/#create-a-zone-in-the-global-control-plane)
+* [yq installed](https://github.com/mikefarah/yq)
+
+## Transfer resources from the self-managed global control plane to {{site.konnect_short_name}}
+
+1. Make sure the self-managed global control plane is active:
+  ```bash
+  kumactl config control-planes list
+  ```
+  This should return your self-managed global control plane. You can use port-forward of the On-Prem Global CP API port 5681 to localhost to get quick access with `kumactl`.
```
Comment (cloudjumpercat) [inline app/konnect/mesh-manager/migrate-zone.md:21]:
```suggestion
  This should return your self-managed global control plane. If you are using Kubernetes, you can run `kubectl port-forward {resource_name} 5681:localhost` to get quick access with `kumactl`.
```

@lobkovilya I like the idea of keeping the port-forward example. Does the formatting look correct to you? I added the curly brackets because "resource name" is a placeholder. I wasn't sure if there was a port # we could add for localhost or if it would depend on the user's environment.

File: app/konnect/mesh-manager/migrate-zone.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (lahabana) [reply app/konnect/mesh-manager/migrate-zone.md:68]: So I think the wording is wrong. Also this zone should have the same name as the existing one.

**Comment by lahabana** [comment]
> Why it would be hard? I can probably add the section at the bottom of the page no?
Sorry typo I meant "add" :) I think it should towards the top of the page and we probably want 2 diagrams one before and one after the migration to make it clear what this does

**Comment by cloudjumpercat** [comment]
@lahabana @lobkovilya If you could add info about the CP config update, restarting the zone, a section about limitations, and attach 2 diagram mockups (pre and post migration), I can revise the text and add our branding to the diagrams.

File: app/konnect/mesh-manager/migrate-zone.md (hunk 8 above)
Comment (lobkovilya) [reply app/konnect/mesh-manager/migrate-zone.md:21]: The command looks like this:
```bash
$ kubectl port-forward deployment/{{ site.mesh_cp_name}} -n {{ site.mesh_namespace }} 5681
```

File: app/konnect/mesh-manager/migrate-zone.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+title: Migrate a self-managed zone control plane to {{site.konnect_saas}}
+content_type: tutorial
+---
+
+If you already have zone control planes in {{site.mesh_product_name}}, you can migrate them to {{site.konnect_short_name}} in Mesh Manager. [Mesh Manager](/konnect/mesh-manager/) allows you to create, manage, and view your {{site.mesh_product_name}} service meshes using the {{site.konnect_short_name}} platform.
+
+Here are a few benefits of creating a mesh deployment in {{site.konnect_short_name}} instead of using a self-managed setup:
+
+* **Kong-managed global control plane:** By creating your mesh in {{site.konnect_short_name}}, your global control plane is managed by Kong. 
+* **All entities in one place:** You can view all your information, such as entities from Kong Ingress Controller (KIC) for Kubernetes, {{site.konnect_short_name}}-managed entities, and now service mesh data, all from one central platform.
```
Comment (cloudjumpercat) [inline app/konnect/mesh-manager/migrate-zone.md:11]:
```suggestion
* **All entities in one place:** You can view all your information, such as entities from {{site.kic_product_name}} (KIC) for Kubernetes, {{site.konnect_short_name}}-managed entities, and now service mesh data, all from one central platform. 
```

File: app/konnect/mesh-manager/migrate-zone.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+![mesh migration before](/assets/images/diagrams/diagram-mesh-migration-before.png)
+
+> **Figure 1:** Diagram that explains how the {{site.mesh_product_name}} global control plane communicates with zone control planes, services, and data planes before migrating. Each service has an associated data plane proxy and those communicate with the zone control plane. The zone control plane then communicates with the {{site.mesh_product_name}} global control plane. All of these entities are self-managed in {{site.mesh_product_name}}.
+
+![mesh migration after](/assets/images/diagrams/diagram-mesh-migration-after.png)
+
+> **Figure 2:** Diagram that explains how the {{site.konnect_short_name}} global control plane communicates with zone control planes, services, and data planes after migrating. Each service has an associated data plane proxy and those communicate with the zone control plane. The services, data plane proxies, and zone control planes are all self-managed in {{site.mesh_product_name}}. The zone control plane then communicates with the {{site.konnect_short_name}} global control plane, which is managed by Kong.
```
Comment (cloudjumpercat) [inline app/konnect/mesh-manager/migrate-zone.md:22]: @lobkovilya If you wouldn't mind reading these over to make sure these sound accurate, that would be great!

File: app/konnect/mesh-manager/migrate-zone.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+  Be sure to override your existing zone control plane `values.yaml` configuration with the values provided in the {{site.konnect_short_name}} UI wizard.
+
+1. If [KDS TLS](/mesh/latest/production/secure-deployment/certificates/#control-plane-to-control-plane-multizone) is enabled with self-signed certificates, set `kuma.controlPlane.tls.kdsZoneClient.secretName=""` so the zone control plane can use public certificates to authenticate with {{site.konnect_short_name}}:
+  ```bash
+  kumactl install control-plane \
+  --set "kuma.controlPlane.tls.kdsZoneClient.secretName=kds-ca-certs" \
+  | kubectl apply -f -
+  ```
+
+1. Restart zone control plane with the new values. {{site.konnect_short_name}} will automatically start looking for the zone. Once {{site.konnect_short_name}} finds the zone, it will display it.
```
Comment (cloudjumpercat) [inline app/konnect/mesh-manager/migrate-zone.md:100]: @lobkovilya Could we add the command to restart the control plane here? I tried to look it up in our docs but couldn't find it.

File: app/konnect/mesh-manager/migrate-zone.md (hunk 10 above)
Comment (lobkovilya) [reply app/konnect/mesh-manager/migrate-zone.md:22]: I'm not sure the phrase "... are self-managed in Kong Mesh" actually makes sense. "Self-managed" means it's managed by the customer (not by Kong), so I don't think we can say "self-managed in <some_place>". I'd rather drop the clarification and keep only "... are self-managed."

Other than that it's very accurate!

File: app/konnect/mesh-manager/migrate-zone.md (hunk 11 above)
Comment (lobkovilya) [reply app/konnect/mesh-manager/migrate-zone.md:100]: I just checked, in GUI for Kubernetes we already provide the command that restarts CP if it already exists (`helm upgrade --install`):
image

For Universal, it doesn't make sense to provide the command, because there are too many ways customers are running Zone CP, I don't think it is worth trying to cover all the cases.

**Review by lahabana — APPROVED** [review]
LGTM great addition!

**Review by cloudjumpercat — APPROVED** [review]
LGTM! Thanks for creating this great guide, @lobkovilya !

**Comment by lobkovilya** [comment]
Thanks a lot for helping with it, @cloudjumpercat!
